[sarahchow03] iP

sarahchow03 opened the pull request.

sarahchow03 committed:
Add increment Level-0

sarahchow03 committed:
Add increment Level-1

Following the instructions for Week 3, Level-1 of Mong can echo commands
sent to the command line. Mong will exit the program when the user types
the command "bye".

sarahchow03 committed:
Add increment Level-2

Following the instructions for Week 3, Level-2 of Mong allows users to
add items into a list, which can be printed in index order when the
"list" command is entered.

Add items to the list: add items from the command line into a list that
displays when the command "list" is typed out.

Index items in the list: items in the list start from the number "1".

sarahchow03 committed:
Add increment Level-3

Following the instructions for Week 3, Level-3 of Mong introduces a
marking command for users to set items in the command list as completed
or uncompleted.

Create a class of ListItem: use a class to implement the adding,
indexing, marking, and unmarking of each item in the list.

The class will handle the instantiation of each ListItem and setting of
the "isCompleted" variable that is assigned to each ListItem in the
array of ListItems.

sarahchow03 committed:
Refactor ListItem class to Task

To better align class names with their use cases in the iP, ListItem
class has been refactored to Task.

sarahchow03 committed:
Add increment Level-4

Level-4 of Mong supports the tracking of three types of data, namely
Todos, Deadlines, and Events. To implement these changes, we use
Inheritance and Polymorphism.

Enumerators for TaskType: we decipher the first word in the command line
as the TaskType. Currently, Mong supports LIST, BYE, MARK, UNMARK,
EVENT, TODO and DEADLINE as valid task types.

Improve code quality: In line with the Week 4 instructions, we improve
code quality through the use of constants (rather than magic numbers),
and following of coding standards. For example, indentations of switch
statements are in-line and naming standards.

Inheritance and polymorphism: Event, Todo, and Deadlines are child
classes of the Task class. The toString() function takes on a
different form based on the whether the object is a Event, Todo, or
Deadline object.

sarahchow03 committed:
Add increment Level-4

Level-4 of Mong supports the tracking of three types of data, namely
Todos, Deadlines, and Events. To implement these changes, we use
Inheritance and Polymorphism.

Enumerators for TaskType: we decipher the first word in the command line
as the TaskType. Currently, Mong supports LIST, BYE, MARK, UNMARK,
EVENT, TODO and DEADLINE as valid task types.

Improve code quality: In line with the Week 4 instructions, we improve
code quality through the use of constants (rather than magic numbers),
and following of coding standards. For example, indentations of switch
statements are in-line and naming standards.

Inheritance and polymorphism: Event, Todo, and Deadlines are child
classes of the Task class. The toString() function takes on a
different form based on the whether the object is a Event, Todo, or
Deadline object.

sarahchow03 committed:
Merge branch 'master' of https://github.com/sarahchow03/ip

sarahchow03 committed:
Add increment Level-4

Level-4 of Mong supports the tracking of three types of data, namely
Todos, Deadlines, and Events. To implement these changes, we use
Inheritance and Polymorphism.

Enumerators for TaskType: we decipher the first word in the command line
as the TaskType. Currently, Mong supports LIST, BYE, MARK, UNMARK,
EVENT, TODO and DEADLINE as valid task types.

Improve code quality: we improve code quality through the use of
constants (rather than magic numbers), and following of coding
standards. For example, indentations of switch statements are in-line
and the following of naming standards.

Inheritance and polymorphism: Event, Todo, and Deadlines are child
classes of the Task class. The toString() function takes on a
different form based on the whether the object is a Event, Todo, or
Deadline object.

sarahchow03 committed:
Revert "Merge branch 'master' of https://github.com/sarahchow03/ip"

This reverts commit d60d2c7a0f464cd532e38f6b0b4206a4a58ae55e, reversing
changes made to 8ca5d3b98b9f2b66910c94646a8b19a7ec381246.

sarahchow03 committed:
Test using the I/O redirection technique

iaso1774 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task{
```
Include a space in between Task and {

iaso1774 replied:
Add a space in between Task and {

iaso1774 reviewed (commented):
I couldn't find anything wrong in this other than the one thing I commented, nice job!

Ridiculouswifi commented on src/main/java/Mong.java:
```diff
@@ -0,0 +1,153 @@
+                break;
+            case MARK:
+                // the itemIndex is -1 than the input from the user
+                itemIndex = Integer.parseInt(input.split(" ")[1]) - 1;
```
Perhaps put the integer 1 as a constant, same as you have done for your methods. It would help to justify why index 1 is being used to get the itemIndex.

Ridiculouswifi commented on src/main/java/Mong.java:
```diff
@@ -0,0 +1,153 @@
+    public static final int LENGTH_TO = 2;
+    public static final String HORIZONTAL_LINE = "--------------------------------------------------";
+
+    /**
```
I like the documentation throughout the code, it helps to understand the functionality of the methods

Ridiculouswifi commented on src/main/java/Mong.java:
```diff
@@ -0,0 +1,153 @@
+        int endOfTo = input.lastIndexOf("/to") + LENGTH_TO;
+        int startOfTo = input.indexOf("/to");
+        String description = input.substring(endOfCommand, startOfFrom);
+        String from = input.substring(endOfFrom + 2, startOfTo - 1);
```
Took a while for me to understand how the +2 and -1 were derived, perhaps you could add a comment to explain it

Ridiculouswifi commented on src/main/java/TaskType.java:
```diff
@@ -0,0 +1,22 @@
+public enum TaskType {
```
Could you explain the purpose of this file? I'm not sure how it improves the flow of the program.

Ridiculouswifi commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,56 @@
+    private String description;
+    private boolean isCompleted;
+    private int index;
+    public static int currentIndex = 0;
```
Would it be possible to shift this variable to the main code instead? Since that is where the array containing the Tasks is. Unless there is a specific reason for having the currentIndex here

Ridiculouswifi reviewed (commented):
Your code flow is good! I can understand the general functions just through the names, although I think there can be more explanation on the implementation, especially with the use of magic numbers

sarahchow03 committed:
Use exceptions to handle errors

Level-5 of Mong is able to handle errors involving the adding, marking,
unmarking, listing, and other basic command errors.

Exceptions will print out the specific error in the terminal.
MongException represents errors specific to Mong such as wrong
task types and erroneous task formats  .

sarahchow03 committed:
Divide classes into packages

Put all classes into 'mong' package.
Organize related classes into either 'task' or 'exception' package.

sarahchow03 committed:
Update test files with the refactoring of code directory

sarahchow03 committed:
Merge branch 'branch-Level-5'

sarahchow03 committed:
Replace with single class imports

sarahchow03 committed:
Import ArrayList collection

sarahchow03 committed:
Adjust index values for description

sarahchow03 committed:
Refactor Array to ArrayList

The ArrayList class is a resizable-array implementation of the List
interface, providing added functionality of deleting items in the list.

sarahchow03 committed:
Add support for deleting tasks in the list

Users will be able to delete tasks by specifying the index of the item
on the list.

sarahchow03 committed:
Refactor code to use ArrayList rather than Array type

sarahchow03 committed:
Refactor from Array to ArrayList

sarahchow03 committed:
Update cosmetic changes

sarahchow03 committed:
Load .txt file data when Mong starts up

sarahchow03 committed:
Write new saved data to .txt file when chatbot closes successfully

sarahchow03 committed:
Create new directory and file for new start up

sarahchow03 committed:
Merge branch 'branch-Level-6'

sarahchow03 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/mong/Mong.java

sarahchow03 committed:
Fix deleteTask function

thoriritu commented on src/main/java/mong/Mong.java:
```diff
@@ -0,0 +1,358 @@
+     * If the command's first word is "unmark", it will set the isCompleted variable of the Task to false.
+     * If the command "bye" is sent, the program will exit.
+     */
+    public static void addByTask() {
```
The method addByTask() handles parsing commands and deciding which actions to take, but this could be abstracted further for better readability and separation of concerns.

thoriritu commented on src/main/java/mong/Mong.java:
```diff
@@ -0,0 +1,358 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
```
Have multiple classes for better abstraction. This is handling storage and task operations as well.

thoriritu commented on src/main/java/mong/Mong.java:
```diff
@@ -0,0 +1,358 @@
+            list.get(itemIndex).setCompleted(true);
+            System.out.println("Mong >_<!!I have marked it as completed:");
+            System.out.println(list.get(itemIndex));
+        } catch (NullPointerException | IndexOutOfBoundsException e) {
```
Currently, the mark and unmark methods throw a NullPointerException or IndexOutOfBoundsException when the index is invalid. It would be better to handle these cases before calling the setCompleted method to improve clarity.

thoriritu commented on src/main/java/mong/Mong.java:
```diff
@@ -0,0 +1,358 @@
+     * Add a Task of type Deadline.
+     * A deadline consists of the task description and deadline in String format.
+     */
+    public static void addDeadline(String input) {
```
The methods addTodo, addDeadline, and addEvent have duplicated logic. Refactor these methods to share common code for validation and task creation to avoid duplication.

sarahchow03 committed:
Update .txt file to save when task list changes

sarahchow03 committed:
Refactor list function

sarahchow03 committed:
Ignore manifest file

sarahchow03 committed:
Fix error handling for deleteTask, mark and unmark functions

sarahchow03 committed:
Add Ui class

The Ui class handles interactions with the users.

sarahchow03 committed:
Add Parser class file

The Parser class makes sense of user inputs.

sarahchow03 committed:
Add TaskList class

The TaskList class contains the list and list operation (add, delete
task).

sarahchow03 committed:
Add Storage class

The Storage class handles loading tasks from the files and saving tasks
to the file.

sarahchow03 committed:
Refactor Main class

sarahchow03 committed:
Merge branch 'branch-More-OOP'

sarahchow03 committed:
Add find function

The find function allows Mong to search for items that contain a single
keyword. The output is a indexed list.

sarahchow03 committed:
Merge pull request #1 from sarahchow03/branch-Level-9

Pull request for branch level 9

sarahchow03 committed:
Refactor import names

sarahchow03 committed:
Update README.md

sarahchow03 committed:
Add User Guide

sarahchow03 committed:
Merge branch 'master' of https://github.com/sarahchow03/ip

sarahchow03 committed:
Add more functionality to find function

The find function searches for a key phrase in this iteration.

sarahchow03 committed:
Add set-up details JavaDoc

sarahchow03 committed:
Merge branch 'branch-A-JavaDoc'

sarahchow03 committed:
Add JavaDoc comments to public methods

sarahchow03 committed:
Refactor code with setTaskCompletion method

sarahchow03 committed:
Update README.md
